### PR TITLE
Added python code for Space Jazz in DP_Ranges.mdx

### PR DESCRIPTION
**Pull request by 1a1242**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: content/4_Gold/DP_Ranges.mdx [hunk 1]
```diff
@@ -134,6 +134,24 @@ public class Jazz {
 ```
 
 </JavaSection>
+	
+<PySection>
+
+```python
+s = input()
+n = len(s)
+dp = [[0] * (n + 5) for _ in range(n + 5)]
```
[the remark this answers is not in the document]
Comment (1a1242) [reply content/4_Gold/DP_Ranges.mdx:143]: Just in case to not access the memory not allocated.

[the remark this answers is not in the document]
Comment (1a1242) [reply content/4_Gold/DP_Ranges.mdx]: Changed it to +2

[the remark this answers is not in the document]
Comment (1a1242) [reply content/4_Gold/DP_Ranges.mdx]: It doesn't

[the remark this answers is not in the document]
Comment (1a1242) [reply content/4_Gold/DP_Ranges.mdx]: we cant use n as in loops i am accessing n+1 th index i used n + 2

[the remark this answers is not in the document]
Comment (1a1242) [reply content/4_Gold/DP_Ranges.mdx]: we cant use n as i am accessing nth element so i changed it to n + 1

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
tabs

**Review by ryanchou-dev — COMMENTED** [review]
space out lines 🙏

**Comment by 1a1242** [comment]
Done

**Comment by 1a1242** [comment]
> tabs

done

**Comment by 1a1242** [comment]
> space out lines 🙏

i did not understand what you meant.

**Comment by SansPapyrus683** [comment]
i think he meant to separate segments of your code with new lines & stuff

**Comment by 1a1242** [comment]
> i think he meant to separate segments of your code with new lines & stuff

Oh okay thanks.

File: content/4_Gold/DP_Ranges.mdx [hunk 2]
```diff
@@ -134,6 +134,26 @@ public class Jazz {
...
+s = input()
+n = len(s)
+dp = [[0] * (n + 1) for _ in range(n + 1)]
+
+for i in range(n + 1):
+	for j in range(n - i):
+		dp[j][i + j] = dp[j + 1][i + j] + 1
+		for k in range(j + 1, i + j + 1):
+			if s[k] == s[j]:
+				dp[j][i + j] = min(dp[j][i + j], dp[j + 1][k - 1] + dp[k + 1][i + j])
```
Comment (SansPapyrus683) [inline content/4_Gold/DP_Ranges.mdx:150]:
```suggestion
				dp[j][i + j] = min(
					dp[j][i + j],
					dp[j + 1][k - 1] + dp[k + 1][i + j]
				)
```

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**